core/iwasm/common/wasm_c_api.c

```diff
@@ -1902,10 +1902,39 @@ wasm_frame_func_offset(const wasm_frame_t *frame)
     return frame ? frame->func_offset : 0;
 }
 
+void
+wasm_frame_vec_clone_internal(wasm_frame_vec_t *src, wasm_frame_vec_t *out)
```

Comment:
would it make sense to implement a generic `bh_vector_copy` and move it to `bh_vector.{c|h}`?

Reply:
In principle I should be able to use https://github.com/bytecodealliance/wasm-micro-runtime/blob/6dbfeb25dd164c0ffcec21806e1c1cd0dff27c58/core/iwasm/common/wasm_c_api.c#L226 but I was having some problems accessing the data https://github.com/bytecodealliance/wasm-micro-runtime/pull/2908#issuecomment-1860794770

Reply:
Ah yes, you should be able to use that function instead. Can you please debug the code step by step then? Quickly looking at the code it seems like it does the same thing, so would be great to re-use it. Please note that the order of parameters in the vec_copy is swapped (i.e. `out` is a first parameter).